sycl/plugins/level_zero/pi_level_zero.cpp

```diff
@@ -482,6 +488,16 @@ pi_result piPlatformsGet(pi_uint32 NumEntries, pi_platform *Platforms,
     assert(ZeDriverCount == 1);
     ZE_CALL(zeDriverGet(&ZeDriverCount, &ZeDriver));
 
+    for (uint32_t i = 0; i < piPlatformsCache.size(); i++) {
```

Comment:
This looks like an STL algorithm

Reply:
@keryell , could you be more specific?

Reply:
https://en.cppreference.com/w/cpp/algorithm/find
https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#p3-express-intent

Reply:
Ok, I changed to the range based loop.